common/flatpak-dir.c

```diff
@@ -1929,6 +1932,7 @@ repo_pull_one_dir (OstreeRepo          *self,
       collection_refs_to_fetch[0] = &collection_ref;
       collection_refs_to_fetch[1] = NULL;
 
+      /* TODO: This implies the summary is downloaded elsewhere. Is this option actually accepted? */
```

Comment:
Yes, we already did this when looking for updates. So we avoid ensure we get exactly the same version we detected then, as various decisions (like permissons) may have been done based on the metadata for that commit.